src/aspire/source/coordinates.py

```
@@ -0,0 +1,485 @@
+        :particle_size: Desired size of cropped particles (will override the size specified in coordinate file).
+        :param max_rows: Maximum number of particles to read. (If `None`, will
+        attempt to load all particles)
+        :param dtype: dtype with which to load images (default: double)
+        """
+        # get full filepaths and data folder
+        mrc_paths, coord_paths = self._check_and_get_paths(files)
+        # instantiate super
+        CoordinateSource.__init__(
+            self, mrc_paths, coord_paths, particle_size, max_rows, dtype
+        )
+
+    def populate_particles(self, mrc_paths, coord_paths):
+        """
+        Extract coordinates from .coord files, which specify particles by the
+        coordinates of their centers.
+        """
+        self._populate_particles(mrc_paths, coord_paths)
+
+    def coords_list_from_file(self, coord_file):
+        """
+        Given a coordinate file with (x,y) particle centers,
+        return a list of coordinates in our canonical (.box) format
+        """
+        # check if it's a STAR file list of centers
+        if os.path.splitext(coord_file)[1] == ".star":
+            return self.coords_list_from_star(coord_file)
+        # otherwise we assume text file format with one coord per line:
+        with open(coord_file, "r") as infile:
+            lines = [line.split() for line in infile.readlines()]
+        # subtract off half of particle size from center coord
+        # populate final two coordinates with the particle_size
+        return [
+            list(map(lambda x: int(x) - self.particle_size // 2, line[:2]))
+            + [self.particle_size] * 2
+            for line in lines
+        ]
+
+
+class RelionCoordinateSource(CoordinateSource):
+    """
+    Represents a data source derived from an autopick.star file within a Relion
+    project directory.
+    """
+
+    def __init__(
+        self, relion_autopick_star, particle_size, max_rows=None, dtype="double"
```

Just checking again whether we are actually inferring particle_size from somewhere (and then this should have a default), or if the docstring is misleading.

---

You're right, `particle_size` must be specified. Docstring fixed